search-app/src/cmr/search/services/parameters/converters/shapefile.clj

```diff
@@ -188,6 +194,29 @@
           (throw e) ;; This was a more specific service error so just re-throw it
           (errors/throw-service-error :bad-request "Failed to parse GeoJSON file"))))))
 
+(defn kml->conditions-vec
+  "Converts a kml file to a vector of SpatialConditions"
+  [shapefile-info]
+  (try
+    (let [file (:tempfile shapefile-info)
+          input-stream (FileInputStream. file)
+          parser (PullParser. (KMLConfiguration.) input-stream SimpleFeature)]
+      (try
+        (loop [conditions []]
+          (if-let [feature (.parse parser)]
+            (let [feature-conditions (feature->conditions feature {})]
+              (if (> (count feature-conditions) 0)
+                (recur (conj conditions (gc/or-conds feature-conditions)))
+                (recur conditions)))
+            conditions))
+        (finally (do
+                  (.delete file)))))
+    (catch Exception e
+      (let [{:keys [type errors]} (ex-data e)]
+        (if (and type errors)
+          (throw e) ;; This was a more specific service error so just re-throw it
+          (errors/throw-service-error :bad-request "Failed to parse KML file"))))))
```

Comment:
You are swallowing `e` here. I agree that returning a generic 400 with failed to parse KML file might make sense, but you should definitely log `e` so that we have logs that say why it failed to parse the KML file. Something like:
```
(do
  (warn e)
  (errors/throw-....)
)
```